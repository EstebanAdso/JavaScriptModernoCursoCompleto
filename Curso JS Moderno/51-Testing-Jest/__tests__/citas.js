import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', () =>{

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () =>{
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Juan',
            telefono: '32432432',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'solo duerme'
        };
        citas.agregarCita(citaObj);

        //prueba que se guarde en spanshop correctamente el objeto
        expect(citas).toMatchSnapshot();
    });

    test('actualizar cita',  () =>{
        const citaActualizada = {
            id,
            mascota: 'Hook2',
            propietario: 'Juan2',
            telefono: '3343243',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'solo duerme y no come'
        };

        citas.editarCita(citaActualizada)

        //Actualizando cita
        expect(citas).toMatchSnapshot();
    })

    test('eliminar cita', () =>{
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot
    })
})