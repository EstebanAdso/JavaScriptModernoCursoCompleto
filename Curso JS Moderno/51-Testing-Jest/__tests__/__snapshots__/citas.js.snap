// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1736647660528,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "solo duerme",
      "telefono": "32432432",
    },
  ],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1736647660528,
      "mascota": "Hook2",
      "propietario": "Juan2",
      "sintomas": "solo duerme y no come",
      "telefono": "3343243",
    },
  ],
}
`;
